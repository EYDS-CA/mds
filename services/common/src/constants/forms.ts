export enum FORM {
  // projects
  ADD_EDIT_PROJECT_SUMMARY = "ADD_EDIT_PROJECT_SUMMARY",
  ADD_MINE_MAJOR_APPLICATION = "ADD_MINE_MAJOR_APPLICATION",
  // reports
  VIEW_EDIT_REPORT = "VIEW_EDIT_REPORT",
  // permits
  REVOKE_DIGITAL_CREDENTIAL = "REVOKE_DIGITAL_CREDENTIAL",
  REQUEST_REPORT = "REQUEST_REPORT",
  ADD_REPORT_TO_PERMIT_CONDITION = "ADD_REPORT_TO_PERMIT_CONDITION",
  ADD_PERMIT_CONDITION_CATEGORY = "ADD_PERMIT_CONDITION_CATEGORY",
  EDIT_PERMIT_CONDITION = "EDIT_PERMIT_CONDITION",
  // tailings
  ADD_TAILINGS = "ADD_TAILINGS",
  // documents
  REPLACE_DOCUMENT = "REPLACE_DOCUMENT",
  DELETE_DOCUMENT = "DELETE_DOCUMENT",
  ARCHIVE_DOCUMENT = "ARCHIVE_DOCUMENT",
  EDIT_HELP_GUIDE = "EDIT_HELP_GUIDE",
  INLINE_EDIT_PERMIT_CONDITION_CATEGORY = "INLINE_EDIT_PERMIT_CONDITION_CATEGORY",
  PERMIT_CONDITION_REVIEW_ASSIGNMENT = "PERMIT_CONDITION_REVIEW_ASSIGNMENT",
  UPDATE_MAJOR_MINE_APPLICATION = "UPDATE_MAJOR_MINE_APPLICATION",
  UPDATE_IRT = "UPDATE_IRT"
}
