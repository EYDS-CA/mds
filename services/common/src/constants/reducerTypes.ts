// Misc
export const AUTHENTICATION = "AUTHENTICATION";
export const STATIC_CONTENT = "STATIC_CONTENT";
export const MODAL = "MODAL";
export const DOCUMENT_VIEWER = "DOCUMENT_VIEWER";
export const SYSTEM_FLAG = "SYSTEM_FLAG";

// Mines
export const MINES = "MINES";

// Parties
export const PARTIES = "PARTIES";

// OrgBook
export const ORGBOOK = "ORGBOOK";

// Compliance Orders
export const COMPLIANCE = "COMPLIANCE";

// MineSpace
export const MINESPACE = "MINESPACE";

// Permits
export const PERMITS = "PERMITS";

// Explosive Storage & Use Permits
export const EXPLOSIVES_PERMITS = "EXPLOSIVES_PERMITS";

// Search
export const SEARCH = "SEARCH";

// Variances
export const VARIANCES = "VARIANCES";

// Projects
export const PROJECTS = "PROJECTS";
export const PROJECT_SUMMARIES = "PROJECT_SUMMARIES";
export const INFORMATION_REQUIREMENTS_TABLE = "INFORMATION_REQUIREMENTS_TABLE";

// Core Users
export const USERS = "USERS";

// Incidents
export const INCIDENTS = "INCIDENTS";

// Work Information
export const MINE_WORK_INFORMATIONS = "MINE_WORK_INFORMATIONS";

// Reports
export const REPORTS = "REPORTS";

// Notices of Work
export const NOTICE_OF_WORK = "NOTICE_OF_WORK";

// Securities
export const SECURITIES = "SECURITIES";

// EPIC Information
export const EXTERNAL_AUTHS = "EXTERNAL_AUTHS";

// Notices of Departure
export const NOTICES_OF_DEPARTURE = "NOTICES_OF_DEPARTURE";

// Activities
export const ACTIVITIES = "ACTIVITIES";

// Tailings Storage Facilities
export const TAILINGS = "TAILINGS";

// Dams
export const DAMS = "DAMS";


//Verifiable Credentials
export const VERIFIABLE_CREDENTIALS = "VERIFIABLE_CREDENTIALS";
