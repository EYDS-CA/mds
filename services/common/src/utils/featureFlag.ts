import { detectProdEnvironment as IN_PROD } from "./environmentUtils";
import flagsmith from "flagsmith";

// Name of feature flags. These correspond to feature flags defined in flagsmith.
export enum Feature {
  MAJOR_PROJECT_ARCHIVE_FILE = "major_project_archive_file",
  DOCUMENTS_REPLACE_FILE = "major_project_replace_file",
  MAJOR_PROJECT_ALL_DOCUMENTS = "major_project_all_documents",
  MAJOR_PROJECT_DECISION_PACKAGE = "major_project_decision_package",
  FLAGSMITH = "flagsmith",
  TSF_V2 = "tsf_v2",
}

// Definition of legacy flags
// TODO: Remove after flagsmith is live
const Flags = {
  [Feature.MAJOR_PROJECT_ARCHIVE_FILE]: !IN_PROD(),
  [Feature.DOCUMENTS_REPLACE_FILE]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_ALL_DOCUMENTS]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_DECISION_PACKAGE]: !IN_PROD(),
  [Feature.TSF_V2]: !IN_PROD(),
  [Feature.FLAGSMITH]: !IN_PROD(),
};

export const initializeFlagsmith = async (flagsmithUrl, flagsmithKey) => {
  await flagsmith.init({
    api: flagsmithUrl,
    environmentID: flagsmithKey,
    cacheFlags: true,
    enableAnalytics: true,
  });
};

/**
 * Returns true if the given feature is enabled
 * @param feature Feature to verify
 * @returns true if the given feature is enabled
 */
export const isFeatureEnabled = (feature: Feature) => {
  if (Feature.FLAGSMITH in Flags && Flags[Feature.FLAGSMITH]) {
    return flagsmith.hasFeature(feature);
  }

  if (feature in Flags) {
    return Flags[feature];
  }

  return false;
};
