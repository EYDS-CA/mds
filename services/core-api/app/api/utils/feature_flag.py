from enum import Enum

from app.config import Config
from flagsmith import Flagsmith
from flask import current_app


class Feature(Enum):
    TSF_V2 = 'tsf_v2'
    MAJOR_PROJECT_REPLACE_FILE = 'major_project_replace_file'
    MINE_APPLICATION_FILE_UDPATE_ALERTS = 'mine_application_file_update_alerts'
    TRACTION_VERIFIABLE_CREDENTIALS = 'verifiable_credentials'
    #if enabled the credential offer will be the current development state of all the 2.0 changes, Q1 2024
    VC_ANONCREDS_20 = 'vc_mines_act_permit_20'   # pending anoncred content differences.
    VC_ANONCREDS_CORE = "vc_anoncreds_core"
    VC_ANONCREDS_MINESPACE = "vc_anoncreds_minespace"
    VC_W3C = "vc_w3c"
    CODE_REQUIRED_REPORTS = 'code_required-reports'
    PERMIT_DOCUMENT_KEYWORD_SEARCH = 'permit_document_keyword_search'
    AMS_AGENT = 'ams_agent'

    def __str__(self):
        return self.value


flagsmith = Flagsmith(
    environment_key=Config.FLAGSMITH_KEY,
    api_url=Config.FLAGSMITH_URL,
)


def is_feature_enabled(feature):
    try:
        feature = str(feature).strip()
        flags = flagsmith.get_environment_flags()

        return feature in flags.flags and flags.is_feature_enabled(feature)
    except Exception as e:
        current_app.logger.error(f'Failed to look up feature flag for: {feature}. ' + str(e))
        return False
