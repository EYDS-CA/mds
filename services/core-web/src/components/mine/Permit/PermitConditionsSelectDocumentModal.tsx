import { IPermitAmendmentDocument } from "@mds/common/interfaces";
import React, { FC } from "react";
import CoreTable from "@mds/common/components/common/CoreTable";
import {
  renderDateColumn,
  renderTextColumn,
} from "@mds/common/components/common/CoreTableCommonColumns";
import { Button, Typography } from "antd";
import { documentNameColumn } from "@mds/common/components/documents/DocumentColumns";

const { Title, Paragraph } = Typography;

interface PermitConditionsSelectDocumentModalProps {
  documents: IPermitAmendmentDocument[];
  onSubmit: (document: IPermitAmendmentDocument) => void;
}

const PermitConditionsSelectDocumentModal: FC<PermitConditionsSelectDocumentModalProps> = ({
  documents,
  onSubmit,
}) => {
  return (
    <div>
      <Title>Extract Permit Conditions</Title>
      <Paragraph>Select the permit for permit condition extraction.</Paragraph>
      <CoreTable
        rowKey="document_manager_guid"
        dataSource={documents}
        columns={[
          documentNameColumn("document_name", "File Name"),
          renderTextColumn("create_user", "Created By"),
          renderDateColumn("create_timestamp", "Uploaded"),
          {
            key: "action",
            className: "actions-column",
            width: 0,
            render: (record: IPermitAmendmentDocument) => {
              return (
                <Button type="primary" onClick={() => onSubmit(record)}>
                  Extract
                </Button>
              );
            },
          },
        ]}
      />
    </div>
  );
};

export default PermitConditionsSelectDocumentModal;
