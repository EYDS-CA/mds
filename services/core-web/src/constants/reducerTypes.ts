// Documents
export const DOCUMENTS = "DOCUMENTS";

// Alerts
export const MINE_ALERTS = "MINE_ALERTS";