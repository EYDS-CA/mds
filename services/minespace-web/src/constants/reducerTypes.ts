// User Dashboard Info
export const USER_MINE_INFO = "USER_MINE_INFO";
