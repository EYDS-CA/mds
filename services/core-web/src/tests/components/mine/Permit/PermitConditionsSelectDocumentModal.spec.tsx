import React from "react";
import { render } from "@testing-library/react";
import * as MOCK from "@mds/common/tests/mocks/dataMocks";
import { ReduxWrapper } from "@mds/common/tests/utils/ReduxWrapper";
import { AUTHENTICATION, MINES, PERMITS, STATIC_CONTENT } from "@mds/common/constants/reducerTypes";
import { USER_ROLES } from "@mds/common/constants/environment";
import { BrowserRouter } from "react-router-dom";
import PermitConditionsSelectDocumentModal from "@/components/mine/Permit/PermitConditionsSelectDocumentModal";

const initialState = {
  [PERMITS]: { permits: MOCK.PERMITS, permitAmendments: MOCK.PERMIT_AMENDMENT_STATE },
  [MINES]: { mines: MOCK.MINES.mines },
  [AUTHENTICATION]: {
    userAccessData: [USER_ROLES.role_admin, USER_ROLES.role_edit_template_conditions],
  },
  [STATIC_CONTENT]: {
    ...MOCK.BULK_STATIC_CONTENT_RESPONSE,
  },
};

function mockFunction() {
  const original = jest.requireActual("react-router-dom");
  return {
    ...original,
    useParams: jest.fn().mockReturnValue({
      id: MOCK.MINES.mineIds[0],
      permitGuid: MOCK.PERMITS[0].permit_guid,
    }),
  };
}

jest.mock("react-router-dom", () => mockFunction());

describe("PermitConditionsSelectDocumentModal", () => {
  it("renders properly", () => {
    const { container } = render(
      <ReduxWrapper initialState={initialState}>
        <BrowserRouter>
          <PermitConditionsSelectDocumentModal documents={MOCK.PERMITS[0].permit_amendments[0].related_documents} onSubmit={jest.fn()}/>
        </BrowserRouter>
      </ReduxWrapper>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
