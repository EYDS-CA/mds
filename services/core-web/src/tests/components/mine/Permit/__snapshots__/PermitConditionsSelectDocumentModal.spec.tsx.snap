// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PermitConditionsSelectDocumentModal renders properly 1`] = `
<div>
  <h1
    class="ant-typography"
  >
    Extract Permit Conditions
  </h1>
  <div
    class="ant-typography"
  >
    Select the permit for permit condition extraction.
  </div>
  <div
    class="ant-table-wrapper  core-table"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
            >
              <table
                style="table-layout: auto;"
              >
                <colgroup />
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      aria-label="File Name"
                      class="ant-table-cell ant-table-column-has-sorters"
                      tabindex="0"
                    >
                      <div
                        class="ant-table-column-sorters"
                      >
                        <span
                          class="ant-table-column-title"
                        >
                          File Name
                        </span>
                        <span
                          class="ant-table-column-sorter ant-table-column-sorter-full"
                        >
                          <span
                            class="ant-table-column-sorter-inner"
                          >
                            <span
                              aria-label="caret-up"
                              class="anticon anticon-caret-up ant-table-column-sorter-up"
                              role="presentation"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-up"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                />
                              </svg>
                            </span>
                            <span
                              aria-label="caret-down"
                              class="anticon anticon-caret-down ant-table-column-sorter-down"
                              role="presentation"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="caret-down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="0 0 1024 1024"
                                width="1em"
                              >
                                <path
                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                />
                              </svg>
                            </span>
                          </span>
                        </span>
                      </div>
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Created By
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Uploaded
                    </th>
                    <th
                      class="ant-table-cell actions-column"
                    />
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row ant-table-row-level-0 fade-in"
                    data-row-key="64caef0e-060d-4875-a470-6c225b242723"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      <div
                        class="inline-flex flex-between file-name-container"
                        title="File Name"
                      >
                        <div
                          class="file-name-text"
                        >
                          <a
                            class="ant-dropdown-trigger ant-dropdown-link"
                            role="button"
                            tabindex="0"
                          >
                            Adams_amendment_1.pdf
                             
                            <span
                              aria-label="down"
                              class="anticon anticon-down"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </a>
                        </div>
                        <span
                          class="file-history-container"
                        />
                      </div>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <div
                        class="create_user-column"
                        title="Created By"
                      >
                        N/A
                      </div>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <div
                        title="Uploaded"
                      >
                        N/A
                      </div>
                    </td>
                    <td
                      class="ant-table-cell actions-column"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="button"
                      >
                        <span>
                          Extract
                        </span>
                      </button>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0 fade-in"
                    data-row-key="64caef0e-060d-4875-a470-6c225b242723"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      <div
                        class="inline-flex flex-between file-name-container"
                        title="File Name"
                      >
                        <div
                          class="file-name-text"
                        >
                          <a
                            class="ant-dropdown-trigger ant-dropdown-link"
                            role="button"
                            tabindex="0"
                          >
                            Adams_amendment_2.pdf
                             
                            <span
                              aria-label="down"
                              class="anticon anticon-down"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                          </a>
                        </div>
                        <span
                          class="file-history-container"
                        />
                      </div>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <div
                        class="create_user-column"
                        title="Created By"
                      >
                        N/A
                      </div>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <div
                        title="Uploaded"
                      >
                        N/A
                      </div>
                    </td>
                    <td
                      class="ant-table-cell actions-column"
                    >
                      <button
                        class="ant-btn ant-btn-primary"
                        type="button"
                      >
                        <span>
                          Extract
                        </span>
                      </button>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
